### ISSUE TITLE
Error: Left and top must be nonnegative

### ISSUE THREAD

**Issue by k4driano** [DantSu/PrinterThermal-ESCPOS-Android#17]
Code:

```java
try {
    Printer printer = new Printer(BluetoothPrinters.selectFirstPairedBluetoothPrinter(), 203, 48f, 32);
    printer.printFormattedText("[C]<qrcode>https://www.example.com/test?p=91681501911252000000000224193018065001689317|2|1|1|ADC97CE801B1C2E864B3EBE92B2E2B5EA87919EF</qrcode>");
    printer.disconnectPrinter();
} catch (Exception ex) {
    Log.e("print", ex.getMessage());
}
```
Error in log: **Left and top must be nonnegative**

**Comment by DantSu** [comment]
Can you paste the full log error ?

**Comment by DantSu** [comment]
I think your adresse is too long for the QRCode size. Change the QRCode size.

**Comment by k4driano** [comment]
This was done and did not resolve.
Version used: 1.1.1
Papper width: 48mm

**Comment by DantSu** [comment]
Paste the full log error.
Your URL is very long. Try `<qrcode size='40'>...</qrcode>`.
Is a very long URL, I don't know if you can do it.

**Comment by k4driano** [comment]
Log error:
```java
java.lang.IllegalArgumentException: Left and top must be nonnegative
        at com.google.zxing.common.BitMatrix.setRegion(BitMatrix.java:233)
        at com.google.zxing.qrcode.QRCodeWriter.renderResult(QRCodeWriter.java:110)
        at com.google.zxing.qrcode.QRCodeWriter.encode(QRCodeWriter.java:79)
        at com.dantsu.printerthermal_escpos_bluetooth.PrinterCommands.QRCodeDataToBytes(PrinterCommands.java:116)
        at com.dantsu.printerthermal_escpos_bluetooth.textparser.PrinterTextParserQRCode.initConstructor(PrinterTextParserQRCode.java:23)
        at com.dantsu.printerthermal_escpos_bluetooth.textparser.PrinterTextParserQRCode.<init>(PrinterTextParserQRCode.java:30)
        at com.dantsu.printerthermal_escpos_bluetooth.textparser.PrinterTextParserColumn.appendQRCode(PrinterTextParserColumn.java:262)
        at com.dantsu.printerthermal_escpos_bluetooth.textparser.PrinterTextParserColumn.<init>(PrinterTextParserColumn.java:73)
        at com.dantsu.printerthermal_escpos_bluetooth.textparser.PrinterTextParserLine.<init>(PrinterTextParserLine.java:42)
        at com.dantsu.printerthermal_escpos_bluetooth.textparser.PrinterTextParser.parse(PrinterTextParser.java:167)
        at com.dantsu.printerthermal_escpos_bluetooth.Printer.printFormattedText(Printer.java:129)
        at br.com.celtasistemas.celtabluetoothprint.MainActivity.imprimir(MainActivity.java:127)
        at br.com.celtasistemas.celtabluetoothprint.MainActivity.access$200(MainActivity.java:25)
        at br.com.celtasistemas.celtabluetoothprint.MainActivity$4.onResponse(MainActivity.java:99)
        at br.com.celtasistemas.celtabluetoothprint.MainActivity$4.onResponse(MainActivity.java:95)
        at com.android.volley.toolbox.StringRequest.deliverResponse(StringRequest.java:82)
        at com.android.volley.toolbox.StringRequest.deliverResponse(StringRequest.java:29)
        at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:102)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:7000)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)

**Comment by DantSu** [comment]
it's the `xzing` library bug. I have to recode this part without use this part : 
```java
       try {
            EnumMap<EncodeHintType, Object> hints = new EnumMap<EncodeHintType, Object>(EncodeHintType.class);
            hints.put(EncodeHintType.CHARACTER_SET, "UTF-8");
            hints.put(EncodeHintType.MARGIN, -1);

            QRCodeWriter writer = new QRCodeWriter();
            bitMatrix = writer.encode(data, BarcodeFormat.QR_CODE, size, size, hints);
        } catch (WriterException e) {
            e.printStackTrace();
        }
```
I have read the code of `xzing` library and I will use `Encoder.encode(data, ErrorCorrectionLevel.L, hints)` but it's a bit more complicated.

**Comment by DantSu** [comment]
Problem solved with [v1.1.2](https://github.com/DantSu/PrinterThermal-ESCPOS-Android/releases/tag/1.1.2)
